pkg/k8shandler/visualization.go

```diff
@@ -283,17 +280,40 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment(prox
 			current, different := isDeploymentDifferent(current, kibanaDeployment)
 
 			// Check trustedCA certs have been updated or not by comparing the hash values in annotation.
-			newTrustedCAHashedValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
+			newTrustedCAHashValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
 			if err != nil {
 				return fmt.Errorf("unable to calculate trusted CA value. E: %s", err.Error())
 			}
-			trustedCAHashedValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
-			if trustedCAHashedValue != newTrustedCAHashedValue {
+			trustedCAHashValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
+			if trustedCAHashValue != newTrustedCAHashValue {
 				different = true
 				if kibanaDeployment.Spec.Template.ObjectMeta.Annotations == nil {
 					kibanaDeployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
 				}
-				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashedValue
+				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashValue
```

Comment:
Does annotation change cause a redeploy?  @bparees ?

Reply:
i believe any change to the podtemplate within the deployment causes a redeploy, but we should verify.

Reply:
yes:
* @igor-karpukhin and i confirmed that in the case where we saw more than one kibana pod the difference between the replica sets was the annotation value
* manually testing this change when I deleted the configmap and forced a new annotation value to be generated i saw a redeploy/update